### ISSUE TITLE
S3 exporter: endpoint documentation / bucket

### ISSUE THREAD

**Issue by dmitryax** [open-telemetry/opentelemetry-collector-contrib#32774]
### Discussed in https://github.com/open-telemetry/opentelemetry-collector-contrib/discussions/32766

<div type='discussions-op-text'>

<sup>Originally posted by **dabcoder** April 30, 2024</sup>
[The S3 exporter documentation](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/awss3exporter/README.md#exporter-configuration) states that the `endpoint` parameter overrides the endpoint used by the exporter instead of constructing it from `region` and `s3_bucket` . 

Yet [in the code](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/e0af2457aed5ba6107da21204eb41d84273ca8dc/exporter/awss3exporter/config.go#L54) there is:

```
if c.S3Uploader.S3Bucket == "" {
  errs = multierr.Append(errs, errors.New("bucket is required"))
}
```

However, when the `endpoint` is provided, it should also contain the S3 bucket, thus I would not expect the `s3_bucket` to have to be provided and this error to be returned. Am I misunderstanding the config setup here? </div>

**Comment by atoulme** [comment]
This is because in some cases (minIO, storJ) I believe the endpoint can be completely different from what s3 expects, and the bucket is dropped. I might be wrong though.

**Comment by atoulme** [comment]
I think this was addressed by #32815. Closing.
